sdk/src/storage/mod.rs

```
@@ -584,13 +584,9 @@ pub(crate) fn try_serde_from_string<T: DeserializeOwned>(value: &str) -> Option<
         let n2 = c2.to_digit(16)?;
         bytes.push((n1 * 16 + n2) as u8);
     }
-    match yazi::decompress(&bytes, yazi::Format::Zlib) {
-        Ok((decompressed, _)) => match postcard::from_bytes(&decompressed) {
-            Ok(v) => Some(v),
-            Err(_) => None,
-        },
-        Err(_) => None,
-    }
+
+    let (decompressed, _) = yazi::decompress(&bytes, yazi::Format::Zlib).unwrap();
+    ciborium::from_reader(std::io::Cursor::new(decompressed)).unwrap()
```

Apologies for the delay, mind returning a `None` instead of unwraping?

---

Are you sure? If it returns None then deserialization errors will fail silently. It was mentioned in the issue that this behavior caused confusion.

---

Maybe use `expect` then

---

Deserialization could fail if you publish a new version of your side with an update structure for your data. The client would have the old version of the serialized data, but the new wasm bundled would try to deserialize the new version of the data.

---

Alright, I'll make it return None for now, and I'll open an issue for error handling